src/Data/Vec/Relation/Binary/Equality/Setoid.agda

```diff
@@ -51,6 +52,14 @@ open PW public using (length-equal)
 ≋-setoid : ℕ → Setoid a (a ⊔ ℓ)
 ≋-setoid = PW.setoid S
 
+------------------------------------------------------------------------
+-- cong
+
+cong-[]≔ : ∀ {n} {xs ys : Vec A n} i p → xs ≋ ys
+  → xs [ i ]≔ p ≋ ys [ i ]≔ p
+cong-[]≔ F.zero _ (_ ∷ eqn) = refl ∷ eqn
+cong-[]≔ (F.suc i) p (x∼y ∷ eqn) = x∼y ∷ (cong-[]≔ i p eqn)
+
```

Comment:
This lemma should move to the place of maximum generality, viz. in this instance `Data.Vec.Relation.Binary.Pointwise.Inductive`, eg as here (with the `Fin n` argument infixed; and the `p` argument could be made implicit?)
```agda
module _ {_∼_ : Rel A ℓ} (refl : Reflexive _∼_) where
  cong-[_]≔ : ∀ {n} i p {xs} {ys} →
               Pointwise _∼_ {n} {n} xs ys → Pointwise _∼_ {n} {n} (xs [ i ]≔ p) (ys [ i ]≔ p)
  cong-[ zero ]≔  p (_ ∷ eqn)   = refl ∷ eqn
  cong-[ suc i ]≔ p (x∼y ∷ eqn) = x∼y ∷ (cong-[ i ]≔ p eqn)
```

For the cost of an additional assumption of `Reflexive` for the underlying relation, so that when re-exported here, and instantiated with the underlying reflexivity of the `Setoid`, yields the lemma you want...

Reply:
I put your definition in Data.Vec.Relation.Binary.Pointwise.Inductive.